src/Symfony/Component/HttpFoundation/JsonResponse.php

```diff
@@ -43,10 +43,14 @@ public function __construct($data = null, $status = 200, $headers = array())
             $data = new \ArrayObject();
         }
 
+        $this->setDefaultEncodingOptions();
+        $this->setData($data);
+    }
+
+    protected function setDefaultEncodingOptions()
```

Comment:
It means that you need to create a sub-class to be able to override the encoding options. What about creating a real setter instead?